source/server/connection_handler_impl.h

```diff
@@ -363,6 +363,11 @@ class ActiveRawUdpListener : public Network::UdpListenerCallbacks,
   void resumeListening() override { udp_listener_->enable(); }
   void shutdownListener() override {
     // The read_filter_ should be deleted before the udp_listener_ is deleted.
+    if (read_filter_.get() != nullptr && udp_listener_.get() == nullptr) {
+      throw Envoy::EnvoyException(
```

Comment:
Does it have to be an exception? If it's just added for the sake of test, I think it's better to written the test in a way that without read_filter_.reset() the test would crash.

Reply:
Ok. I will change that code and the test to as you mentioned. :)